packages/hawtio/src/ui/page/HawtioHeader.tsx

```diff
@@ -109,20 +109,31 @@ const HawtioHeaderToolbar: React.FunctionComponent = () => {
    * Determine which plugin is currently displaying
    * based on the path of the current location
    */
-  const pluginFromLocation = (): Plugin | null => {
+  const filterPlugins = (): Plugin[] => {
     const path = location.pathname
-    return plugins.find(plugin => path.startsWith(plugin.path)) ?? null
+    return plugins.filter(plugin => {
+      if (path.startsWith(plugin.path)) return true
+
+      if (!plugin.headerItems) return false
+
+      return plugin.headerItems.universal === true
+    })
   }
 
-  const plugin = pluginFromLocation()
+  const filteredPlugins = filterPlugins()
 
   return (
     <Toolbar id='hawtio-header-toolbar'>
       <ToolbarContent>
         <ToolbarGroup>
-          {plugin?.headerItems?.map((comp, index) => (
-            <ToolbarItem key={`hawtio-header-toolbar-plugin-item-${index}`}>{React.createElement(comp)}</ToolbarItem>
-          ))}
+          {filteredPlugins.map(
+            plugin =>
+              plugin.headerItems?.components.map((comp, index) => (
```

Comment:
Then you can directly map against the `headerItems` components here instead of the lengthy `filterPlugins.map(plugin => plugin.headerItems?.components)` chain.